core/src/Language/Avaleryar/Syntax.hs

```diff
@@ -59,12 +61,26 @@ import           GHC.Generics                 (Generic)
 import           Text.Megaparsec              (SourcePos(..), pos1, unPos)
 import           Text.PrettyPrint.Leijen.Text
     (Doc, Pretty(..), brackets, colon, dot, empty, group, hsep, line, nest, parens, punctuate, space, vsep)
+import           Text.Read (readMaybe)
 
-data Value
-  = I Int
-  | T Text
-  | B Bool
-    deriving (Eq, Ord, Read, Show, Generic)
+newtype Value = T Text
+  deriving (Eq, Ord, Read, Show, Generic)
+
+-- COMPLETE Pragma is necessary because the exhausiveness checker doesn't work at all with pattern synonyms.
+-- See https://gitlab.haskell.org/ghc/ghc/-/wikis/pattern-synonyms/complete-sigs
+{-# COMPLETE I, B, T #-}
```

Comment:
Same trick that I used when I changed the representation of IP addresses in portal-suite:
https://github.com/Simspace/portal-suite/blob/dev/common-utilities/haskell/src/SimSpace/Utils/Cidr.hs#L36-L43